SeedSigner/seedsigner
# SettingsQR: Final changes before official release

SettingsQR has been quietly supported for a while, but this PR will fully promote SettingsQR to be a user-facing feature.

Two repos have to work hand-in-hand here:
* SeedSigner w/updated SettingsQR support (this PR)
* [SettingsQR Generator](https://github.com/kdmukai/seedsigner-settings-generator)

---
# Changes in this PR:
The expected SettingsQR format has been slightly tweaked.

Parsing them has been greatly simplified.

Missing settings (either in an onboard settings.json file or in a scanned SettingsQR) are now set to their default value. The only exception are hidden settings (e.g. `qr_brightness`); their values are preserved if they are already in the current settings.

---

# How to test
You can use the current public SettingsQR Generator as-is: https://seedsigner.github.io/seedsigner-settings-generator

It is in sync with the changes in this PR.

If you want to build the SettingsQR Generator locally, follow the instructions in [the repo](https://github.com/seedsigner/seedsigner-settings-generator). Just note that there is a dependence on the SeedSigner repo and it is currently pointing to this branch in my fork.

Before the SettingsQR Generator is fully released, we must:
- [x] Transfer ownership to SeedSigner
- [x] Update the developer instructions and the `git` submodule to point to the main SeedSigner repo and its `dev` branch.
